pkg/controller/releasemanager/releasemanager_controller.go

```diff
@@ -0,0 +1,407 @@
+// and what is in the ReleaseManager.Spec
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileReleaseManager) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling ReleaseManager")
+
+	// Fetch the ReleaseManager instance
+	instance := &picchuv1alpha1.ReleaseManager{}
+	if err := r.client.Get(context.TODO(), request.NamespacedName, instance); err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	cluster := &picchuv1alpha1.Cluster{}
+	key := client.ObjectKey{request.Namespace, instance.Spec.Cluster}
+	if err := r.client.Get(context.TODO(), key, cluster); err != nil {
+		return reconcile.Result{}, err
+	}
+	incarnationList := &picchuv1alpha1.IncarnationList{}
+	labelSelector := client.MatchingLabels(map[string]string{
+		"medium.build/app":     instance.Spec.App,
+		"medium.build/cluster": instance.Spec.Cluster,
+	})
+	if err := r.client.List(context.TODO(), labelSelector, incarnationList); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	remoteClient, err := utils.RemoteClient(r.client, cluster)
+	if err != nil {
+		reqLogger.Error(err, "Failed to create remote client", "Cluster.Key", key)
+		return reconcile.Result{}, err
+	}
+	reqLogger.Info("Created remote client", "Cluster.Key", key)
+	manager := ReleaseManager{
+		Instance:        instance,
+		IncarnationList: incarnationList,
+		Cluster:         cluster,
+		Client:          remoteClient,
+		PicchuClient:    r.client,
+	}
+	if err := manager.SyncNamespace(); err != nil {
+		return reconcile.Result{}, err
+	}
+	if err := manager.SyncService(); err != nil {
+		return reconcile.Result{}, err
+	}
+	if err := manager.SyncDestinationRule(); err != nil {
+		return reconcile.Result{}, err
+	}
+	if err := manager.SyncVirtualService(); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{}, nil
+}
+
+type ReleaseManager struct {
+	Instance        *picchuv1alpha1.ReleaseManager
+	IncarnationList *picchuv1alpha1.IncarnationList
+	Cluster         *picchuv1alpha1.Cluster
+	Client          client.Client
+	PicchuClient    client.Client
+}
+
+func (r *ReleaseManager) SyncNamespace() error {
+	labels := map[string]string{
+		"istio-injection": "enabled",
+	}
+	namespace := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   r.Instance.Spec.App,
```

Comment:
what do you think about like `lite-production`?

Reply:
it's per cluster, not fleet

Reply:
I was thinking of production as a target name. might we ever deploy multiple targets of an app on the same fleet?

Reply:
oh, I see what you mean. I actually thought the same thing in bed last night! didn't notice it was namespace on first read. Yeah, we'll do this change in a future PR.